src/components/UserProfile/BasicInformationTab/__tests__/BasicInformationTab.test.js

```diff
@@ -0,0 +1,600 @@
+    expect(formatPhoneNumber("123-456-7890")).toBe("( 123 ) 456 - 7890");
+  });
+
+  it('Test case 6 : Formats phone number with leading or trailing whitespace', () => {
+    expect(formatPhoneNumber("  1234567890  ")).toBe("( 123 ) 456 - 7890");
+  });
+});
+
+/*************************************************************** */
+describe('Test Suite for Phone component', () => {
+  let testProps= {
+   userProfile: { phoneNumber:"123456789", privacySettings:{email :true}},
+   setUserProfile: jest.fn(),
+   formValid: { email: true },
+   setFormValid: jest.fn(),
+   canEdit: true,
+   desktopDisplay: true,
+   handleUserProfile:jest.fn(),
+ };
+
+ it('Test case 1 : Phone component renders with editable fields when canEdit is true ', () => {
+  render(<Phone {...testProps} />);
+   const phInputStyle=screen.queryByText('ph-input-style');
+   expect(phInputStyle).toBeInTheDocument();
+   const phoneToggle=screen.queryByText('phone');
+   expect(phoneToggle).toBeInTheDocument();
+  
+ });
+
+ it('Test case 2 : Phone component renders non-editable fields when canEdit is false ', () => {
+   testProps.canEdit = false;
+   render(<Phone {...testProps} />);
+   expect(screen.queryByText('ph-input-style')).not.toBeInTheDocument();
+   expect(screen.queryByText('phone')).not.toBeInTheDocument();
+ });
+
+
+
+it('Test case 3 : Verify  if phone number  is not displayed if privacy settings and phone is falsey ', () => {
+  testProps= {
+    userProfile: {phoneNumber:'123456789'},
+    setUserProfile: jest.fn(),
+    formValid: { email: true },
+    setFormValid: jest.fn(),
+    canEdit: true,
+    desktopDisplay: true,
+    handleUserProfile:jest.fn(),
+  };  render(<Email {...testProps} />);
+  
+  expect(screen.queryByText((testProps.userProfile.phoneNumber).trim())).not.toBeInTheDocument();
+
+});
+
+});
+
+/******************************************************** */
+
+jest.mock('react-toastify', () => ({
+  toast: {
+    error: jest.fn(),
+  },
+}));
+
+describe('Test suite for TimeZoneDifference component ', () => {
+  beforeEach(() => {
+    jest.useFakeTimers(); // Mock timers
+  });
+
+  afterEach(() => {
+    jest.runOnlyPendingTimers();
+    jest.useRealTimers();
+  });
+
+
+  let testProps = {
+    isUserSelf: false,
+    errorOccurred: false,
+    setErrorOccurred: jest.fn(),
+    desktopDisplay: true,
+    userProfile: {
+      timeZone: 'America/New_York',
+    },
+  };
+
+ 
+
+  it('Test case 1 : Renders "This is your own profile page" when user is self', () => {
+    testProps.isUserSelf=true;
+    render(<TimeZoneDifference {...testProps} />);
+    expect(screen.getByText('This is your own profile page')).toBeInTheDocument();
+  });
+
+  it('Test case 2 : Renders timezone offset when user is not self', () => {
+    testProps.isUserSelf=false;
+    render(<TimeZoneDifference {...testProps} />);
+    expect(screen.getByText(/[-+]\d+ hours/)).toBeInTheDocument(); 
```

Comment:
you can make it optionally by adding `?` before the slash since this test will fail with 0 hours.
so it would be `/[-+]?\d+ hours/`

Reply:
Thank you for the pointer.  I've made that change  ! Kindly retest when you get some time @j-corp-25